### PR TITLE
Feature: improved schema proposition error handling

### PR DESCRIPTION
**Pull request by ryansingman**
- Adds custom errors for:
  - ColumnMismatchError (number of columns in header != number of columns in data)
  - EmptyDatasetError

### CODE REVIEW

**Comment by ryansingman** [comment]
Merge order:
- [CLI](https://github.com/cleanlab/cleanlab-studio/pull/72)- [Frontend](https://github.com/cleanlab/cleanlab-studio-frontend/pull/240)
- [Backend](https://github.com/cleanlab/cleanlab-studio-backend/pull/183)

File: cleanlab_studio/errors.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class InvalidDatasetError(RuntimeError):
+    ...
```
Comment (anishathalye) [inline cleanlab_studio/errors.py:2]: I think it's idiomatic to use `pass` rather than Ellipsis here

File: cleanlab_studio/version.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-__version__ = "0.1.19"
+__version__ = "0.1.20"
```
Comment (anishathalye) [inline cleanlab_studio/version.py:1]: Let's do version bumps separately

File: cleanlab_studio/errors.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class InvalidDatasetError(RuntimeError):
```
Comment (anishathalye) [inline cleanlab_studio/errors.py:1]: Subclass `ValueError` instead? It's more specific and makes sense in this context.

**Review by anishathalye — CHANGES_REQUESTED** [review]
I like the principle of giving more specific error messages in general, though I wouldn't work too hard to support particularly badly formatted datasets like empty ones.

This change seems fine overall, aside from some minor details.

**Comment by ryansingman** [comment]
> I like the principle of giving more specific error messages in general, though I wouldn't work too hard to support particularly badly formatted datasets like empty ones.
> 
> This change seems fine overall, aside from some minor details.

Agreed on not wanting to support egregiously badly formatted datasets. This change is just to allow us to catch and handle those errors properly (not resulting in unhandled exceptions in the backend)

**Review by anishathalye — APPROVED** [review]

**Pull request merged**